src/main_thread/decrypt/__tests__/find_key_system.test.ts

```diff
@@ -0,0 +1,322 @@
+        robustness: "HW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/mp4;codecs="hvc1.1.6.L93.B0"',
+        robustness: "HW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/webm;codecs="vp8"',
+        robustness: "HW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/mp4;codecs="avc1.4d401e"',
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: 'video/mp4;codecs="avc1.42e01e"',
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: 'video/mp4;codecs="hvc1.1.6.L93.B0"',
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: 'video/webm;codecs="vp8"',
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: 'video/mp4;codecs="avc1.4d401e"',
+        robustness: "SW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/mp4;codecs="avc1.42e01e"',
+        robustness: "SW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/mp4;codecs="hvc1.1.6.L93.B0"',
+        robustness: "SW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'video/webm;codecs="vp8"',
+        robustness: "SW_SECURE_CRYPTO",
+      },
+    ],
+    audioCapabilities: [
+      {
+        contentType: 'audio/mp4;codecs="mp4a.40.2"',
+        robustness: "HW_SECURE_ALL",
+      },
+      {
+        contentType: "audio/webm;codecs=opus",
+        robustness: "HW_SECURE_ALL",
+      },
+      {
+        contentType: 'audio/mp4;codecs="mp4a.40.2"',
+        robustness: "HW_SECURE_DECODE",
+      },
+      {
+        contentType: "audio/webm;codecs=opus",
+        robustness: "HW_SECURE_DECODE",
+      },
+      {
+        contentType: 'audio/mp4;codecs="mp4a.40.2"',
+        robustness: "HW_SECURE_CRYPTO",
+      },
+      {
+        contentType: "audio/webm;codecs=opus",
+        robustness: "HW_SECURE_CRYPTO",
+      },
+      {
+        contentType: 'audio/mp4;codecs="mp4a.40.2"',
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: "audio/webm;codecs=opus",
+        robustness: "SW_SECURE_DECODE",
+      },
+      {
+        contentType: 'audio/mp4;codecs="mp4a.40.2"',
+        robustness: "SW_SECURE_CRYPTO",
+      },
+      {
+        contentType: "audio/webm;codecs=opus",
+        robustness: "SW_SECURE_CRYPTO",
+      },
+    ],
+    distinctiveIdentifier: "optional",
+    persistentState: "optional",
+    sessionTypes: ["temporary"],
+  };
+
+  it("should create a media key the first time and then reuse the previous one if it's the same configuration", async () => {
+    requestMediaKeySystemAccessMock.mockImplementation(() => {
+      return {
+        createMediaKeys: () => ({
+          createSession: () => ({
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            generateRequest: () => {},
+          }),
+        }),
+        getConfiguration: () => {
+          return baseEmeConfiguration;
```

Comment:
So here, the returned configuration is unrelated to what has been asked for (e.g. robustnesses) and seems even not really possible

e.g. `distinctiveIdentifier` is returned as `optional`, [something that it cannot be](https://www.w3.org/TR/encrypted-media-2/#dom-mediakeysystemaccess-getconfiguration):
> All values in the configuration may be used in any combination. Members of type [MediaKeysRequirement](https://www.w3.org/TR/encrypted-media-2/#dom-mediakeysrequirement) reflect whether the capability is required for any combination. They will not have the value "[optional](https://www.w3.org/TR/encrypted-media-2/#dom-mediakeysrequirement-optional)".

Anyway we don't care for this test and this is a nitpick